database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -97,7 +100,10 @@ public void onChildChanged(EventType type, DataSnapshot snapshot, int index, int
 
     @Override
     public void onDataChanged() {
-        // No-op, we don't listen to batch events for the key ref
+        if (mHasQueuedInstantOps) {
+            notifyListenersOnDataChanged();
```

Comment:
For an "instant operation" will there be two batch events filed?  I see one here, and one in the DataRefListener.